### ISSUE TITLE
vpa-recommender cannot fetch legacy selector

### ISSUE THREAD

**Issue by vpnachev** [gardener/gardener#1942]
**What happened**:
On a seed cluster:
```bash
$ kubectl -n garden logs vpa-recommender-d8f54f49d-pfmpr --since 1m
...
E0214 20:18:04.913555       1 cluster_feeder.go:476] Error while fetching legacy selector. Reason: v1beta1 selector not found
...
```

This error is logged with very high rate
```bash
$ kubectl -n garden logs vpa-recommender-d8f54f49d-pfmpr --since 10m | grep 'Error\ while\ fetch' -c
8192
```

**What you expected to happen**:
To not see logs of severity `error`, or at least they have to be vary rare.

**How to reproduce it (as minimally and precisely as possible)**:
Run a Gardener landscape with version `v1.0.4` and HVPA featureGate enabled.

**Anything else we need to know?**:

**Environment**:

- Gardener version: v1.0.4
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

**Comment by rfranzke** [comment]
Any ideas?

**Comment by wyb1** [comment]
This error is no longer in the master of branch of vpa. I guess updating should remove this error log. I will check and verify.

**Comment by rfranzke** [comment]
@wyb1 did you have a chance to check? If no, @amshuman-kr @ggaurav10 can you help out?

**Comment by ggaurav10** [comment]
As @wyb1 mentioned, this is removed in latest VPA release.
Please refer: https://github.com/kubernetes/autoscaler/commit/4e07e1eac160759c9c515aa0c7d2db137dd7b06e

Updating the VPA version to 0.7.1 should fix this.
The log error message appears mainly because of wrongly configured log level. We are using VPA `v1beta2`, but VPA first tries to fetch the selector using `v1beta1` selector fetcher, and logs the message if that fails.

**Comment by rfranzke** [comment]
Alright, thanks Gaurav. Can somebody take the task of upgrading VPA then, please? :)

**Comment by rfranzke** [comment]
ping @ggaurav10 @wyb1 @amshuman-kr

**Comment by wyb1** [comment]
I will upgrade vpa to the newest version.

**Comment by rfranzke** [comment]
Thanks a lot!

**Comment by amshuman-kr** [comment]
FYI Another change to VPA that I filed to be able to reduce the recommendation decay half-life for faster scale down has been [merged](https://github.com/kubernetes/autoscaler/pull/2905) a couple of weeks back. Unfortunately, it is not yet released. Once that gets released, we might have to move to 0.8.x or higher.
